zerver/views/push_notifications.py

```diff
@@ -155,4 +167,4 @@ def self_hosting_auth_redirect(request: HttpRequest) -> HttpResponse:  # nocover
 
     redirect_url = result["billing_access_url"]
     assert isinstance(redirect_url, str)
-    return HttpResponseRedirect(redirect_url)
+    return HttpResponseRedirect(redirect_url + f"?next={next}")
```

Comment:
So from a plumbing standpoint, this approach for how `next` is stored is problematic for our ability to update the options without waiting for self-hosters to update their system. I think we want to instead add `next` to `post_data`, and have the bouncer server just return a `redirect_url` that takes into account whatever you passed as `next`.

That way, both the code for what URLs exist and the code for how to handle these `next` values live in the same bouncer codebase.

Reply:
Made this change.